### PR TITLE
SAK-42890 Site Info > Groups: Indicate which tools and items are lock…

### PR DESCRIPTION
**Pull request by mpellicer**
…ing each group.

The 20.x version of this PR is #7977

### CODE REVIEW

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java  4
- site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/MainController.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5134789&bid=16872148)

File: site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/MainController.java [hunk 1]
```diff
@@ -75,11 +75,15 @@ public int compare(Group g1, Group g2){
                 return g1.getTitle().compareToIgnoreCase(g2.getTitle());
         }});
 
-        List<Group> lockedGroupList = site.getGroups().stream().filter(group -> RealmLockMode.ALL.equals(group.getRealmLock()) || RealmLockMode.MODIFY.equals(group.getRealmLock())).collect(Collectors.toList());
-        List<Group> lockedForDeletionGroupList = site.getGroups().stream().filter(group -> RealmLockMode.ALL.equals(group.getRealmLock()) || RealmLockMode.DELETE.equals(group.getRealmLock())).collect(Collectors.toList());
+        // Control the groups that are locked by entities
+        boolean anyGroupLocked = false;
+        List<String> lockedGroupList = new ArrayList<String>();
```
Comment (ern) [inline site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/MainController.java]: can use diamond operator here
ArrayList<>()

File: site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java [hunk 2]
```diff
@@ -42,6 +51,9 @@
 @Slf4j
 public class SakaiService  {
```
Comment (ern) [inline site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java]: super generic and vague name SakaiService, also given that this is in a tool this shouldn't be named a service as it's misleading!

maybe something like:
GroupManagerOperations
GroupManagerActions

File: site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java [hunk 3]
```diff
@@ -126,4 +138,44 @@ public void postEvent(String event, String userId) {
...
+        List<String[]> groupRealmLocks = group.getRealmLocks();
+
+        // Add one list per entity that locks groups.
+        List<String> assignmentTitles = new ArrayList<String>();
+        List<String> assessmentTitles = new ArrayList<String>();
+
+        for (String[] groupRealmLock : groupRealmLocks) {
+            String objectReference = groupRealmLock[0];
+            if (StringUtils.isNotBlank(objectReference)) {
+                if (objectReference.startsWith(GroupManagerConstants.ASN_ENTITY_PREFIX)) {
```
Comment (ern) [inline site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java]: Use the AssignmentConstants in the pi that's the whole purpose so we don't recreate all these constants.

File: site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java [hunk 4]
```diff
@@ -126,4 +138,44 @@ public void postEvent(String event, String userId) {
...
+                    String assignmentReference = objectReference.substring(objectReference.lastIndexOf("/") + 1);
```
Comment (ern) [inline site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java]: Use the AssignmentReferenceReckoner to load an assignment reference which you can easily get the assignment to load

File: site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/MainController.java (hunk 1 above)
Comment (mpellicer) [reply site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/MainController.java]: Well, it's optional right? I like declaring the types but ok, I've changed them.

File: site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java (hunk 2 above)
Comment (mpellicer) [reply site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java]: I don't agree with this, this is a "service that interacts with some Sakai services", SakaiService makes sense to me. I'm not going to change the name of the service and all the variables in this PR.

File: site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java (hunk 3 above)
Comment (mpellicer) [reply site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java]: Ok, thanks for the constant!

File: site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java (hunk 4 above)
Comment (mpellicer) [reply site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java]: Thanks for that, implemented the change, now using the AssignmentReferenceReckoner.

**Comment by mpellicer** [comment]
@ern all the changes have been implemented.

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/MainController.java  3
- site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/service/SakaiService.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5134789&bid=16872148)

**Comment by fostersdesign** [comment]
> @fostersdesign I didn't understand your comment about keyboard nav and tabindex?

@ottenhoff, can you remind me where I said this?

**Pull request merged**